Automattic/woocommerce-payments
# Add EPS as UPE payment method

Fixes #3543 

#### Changes proposed in this Pull Request

* Add EPS to UPE payment methods.
* Update tests accordingly.
* Support discounts for EPS.

Disclaimer: fees are implemented in another PR 1502-gh-Automattic/woocommerce-payments-server.

#### Testing instructions

* Under dev tools the boxes for _Enable grouped settings_, _Enable UPE checkout_, and _Add UPE additional payment methods_ need to be checked.
* If EUR is your default currency, navigate to WooCommerce > Settings > General and change the currency to USD.
* If EUR is an enabled currency in your store, navigate to WooCommerce > Settings > Multi-Currency and remove EUR as a currency.
* Navigate to Payments > Settings.
* Find _Payments accepted on checkout_, verify EPS is in the list, along with its icon.
* Select EPS and click _Save changes_, verify that changes were saved.
* Navigate to WooCommerce > Settings > Multi-Currency and verify EUR is now an enabled currency in your store.
* On the frontend, add an item to your cart, and go to checkout.
* Verify your currency *is not* EUR, payment method should show as _Credit card / debit card_ and when selecting _Use a new payment method_, EPS should not appear.
* Switch your currency to EUR, payment method should show as _Popular payment methods_ and when selecting _Use a new payment method_, EPS should appear.
* Select EPS.
* Complete checkout/place order.
* You will be taken to a Stripe test confirmation page, choose _Authorize test payment_.
* Once order is received, navigate to WooCommerce > Orders in the admin.
* Find your order and select it.
* Click _Refund_, and then increase a line item in your order to 1.
* Click _Refund via WooCommerce Payments_, verify your refund completed.
* Navigate to Payments > Transactions, you should see your _Payment_ and _Payment refund_ listed for your order, verify both have the Bancontact logo.
* Click on the Payment, then verify the _Payment method_ both at the top and bottom are correctly formatting (they will have default test info).
* Do the same for the Payment refund.
* Navigate to Payments > Settings, then click the trash can icon to delete EPS. Verify the name and the logo in the modal are correct, then click _Remove_, scroll to the bottom and _Save changes_.
* Verify EPS is no longer in enabled the list of _Payment methods_.

#### Screenshots
![image](https://cdn-std.droplr.net/files/acc_1180931/EWiMmQ)
![image](https://user-images.githubusercontent.com/79862886/149326429-297b5cc4-731a-4fb7-9904-ae75d72181e2.png)
![image](https://cdn-std.droplr.net/files/acc_1180931/0ZJ8sX)
![image](https://cdn-std.droplr.net/files/acc_1180931/0Zs3Wi)
![image](https://cdn-std.droplr.net/files/acc_1180931/yZRiiD)

-------------------


**Post merge**
